src/test/java/org/folio/rest/impl/ModTenantAPIUnitTest.java

```diff
@@ -0,0 +1,120 @@
+package org.folio.rest.impl;
+
+import static org.folio.rest.impl.OkapiMockServer.OAI_TEST_TENANT;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+
+import org.folio.config.ApplicationConfig;
+import org.folio.oaipmh.common.TestUtil;
+import org.folio.postgres.testing.PostgresTesterContainer;
+import org.folio.rest.jaxrs.model.TenantAttributes;
+import org.folio.rest.persist.PostgresClient;
+import org.folio.spring.SpringContextUtil;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
+import io.vertx.core.http.HttpMethod;
+import io.vertx.core.http.HttpServer;
+import io.vertx.junit5.VertxExtension;
+import io.vertx.junit5.VertxTestContext;
+
+@ExtendWith(VertxExtension.class)
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class ModTenantAPIUnitTest {
+
+  private static final String TABLES_QUERY = "select * from pg_tables where schemaname='" + PostgresClient.convertToPsqlStandard(OAI_TEST_TENANT) + "'";
+  private static final List<String> EXPECTED_TABLES = List.of("set_lb", "instances", "request_metadata_lb", "databasechangelog", "databasechangeloglock");
+
+  private int okapiPort = -1;
+  private ModTenantAPI modTenantAPI;
+  private TenantAttributes tenantAttributes = new TenantAttributes().withModuleTo("99.99.99");
+
+  @BeforeAll
+  void beforeAll(Vertx vertx, VertxTestContext vtc) {
+    var context = vertx.getOrCreateContext();
+    SpringContextUtil.init(vertx, context, ApplicationConfig.class);
+    vertx.runOnContext(v -> modTenantAPI = new ModTenantAPI());
+    PostgresClient.setPostgresTester(new PostgresTesterContainer());
+    PostgresClient.getInstance(vertx, OAI_TEST_TENANT).startPostgresTester();
+    startOkapiMockServer(vertx, vtc);
+  }
+
+  @AfterAll
+  void tearDown(Vertx vertx, VertxTestContext vtc) {
+    vertx.close(res -> {
+      if(res.succeeded()) {
+        vtc.completeNow();
+      } else {
+        vtc.failNow(res.cause());
+      }
+    });
+  }
+
+  private Future<HttpServer> startOkapiMockServer(Vertx vertx, VertxTestContext testContext) {
```

Comment:
```suggestion
  private Future<HttpServer> startOkapiMockServer(Vertx vertx) {
```
This method returns a Future and therefore should not complete the testContext.
Better call like this:
```
startOkapiMockServer(vertx)
.onComplete(vtc.succeedingThenComplete());
```